### DISCUSSION TITLE
Can I run an init container with Aspire?

### DISCUSSION THREAD

**Discussion by Petteroe** [microsoft/aspire#4515] [Q&A]
My scenario is that I have an api project and an EF core migrations project. Same structure as the Aspire samples with a separate project for the migrations for a dbcontext in the api project. These start up, and while migrations are running the api container is failing the connection. No problem, it retries and after migrations are done it starts up.

My question is; could the migrations container be set up as an init container? (https://learn.microsoft.com/en-us/azure/container-apps/containers#init-containers)?

It would be very nice to get the migrations container to spin down after it is done, so that it doesn't sit there and take up unnecessary resources just to answer health check requests.

**Comment by Fazer01** [comment]
Hi there! I had the same question. Apparently there is an issue that's open. See #2866

**Comment by CZEMacLeod** [comment]
I commented on #2866 about this - but have you tried the [WaitFor](https://github.com/davidfowl/WaitForDependenciesAspire) project by @davidfowl  to at least handle the spin up ordering. Not sure about spinning it down on completion though.
